contracts/colonyNetwork/ColonyNetworkDeployer.sol

```diff
@@ -175,6 +178,94 @@ contract ColonyNetworkDeployer is ColonyNetworkStorage {
     // This is intentional, as we want to allow the same Colony to be deployed on different chains
   }
 
+  function setDomainTokenReceiverResolver(address _resolver) public stoppable auth {
+    domainReceiverResolverAddress = _resolver;
+  }
+
+  function getDomainTokenReceiverResolver() public view returns (address) {
+    return domainReceiverResolverAddress;
+  }
+
+  function checkDomainTokenReceiverDeployed(
```

Comment:
Can we choose a different naming convention for this type of idempotent deployment? I feel the same about `deployCreateXIfNeeded`. I'd be happy rolling this up into `getDomainTokenReceiverAddress` with a dev note that it deploys the contract idempotently if needed. `idempotentDeployDomainTokenReceiver` goes in that direction but is just as wordy so I don't know if that's a huge win. Open to your thoughts on this.

Reply:
> I'd be happy rolling this up into getDomainTokenReceiverAddress with a dev note that it deploys the contract idempotently if needed.

I'm quite against this, as it makes it significantly less intuitive to use from the frontend side (as it's no longer a view function).

In terms of naming... I agree that it's much of a muchness. I will say I'm completely unfussed about `deployCreateXIfNeeded` though, as it's a non-contract function only used in tests.

Reply:
Ok, how about `idempotentDeployCONTRACTNAME` for these types of things? So `idempotentDeployCreateX` and `idempotentDeployDomainTokenReceiver`?